### PR TITLE
feat(data-warehouse): implement split_io import source

### PR DESCRIPTION
**Pull request by Gilbert09**
## Problem

Split (now Harness FME) is a feature flagging and experimentation platform with connectors in Airbyte and Fivetran, but PostHog's data warehouse only had a scaffolded stub for it (`unreleasedSource=True`, no sync logic). Users can't pull their Split workspaces, flags, segments, or change requests into PostHog.

## Changes

Implements the `split_io` source end to end, following the `ResumableSource` + `requests` shape of recent sources (LaunchDarkly is the closest analog):

- `settings.py` declares 10 endpoints mirroring the Airbyte/Fivetran stream set: `workspaces`, `environments`, `traffic_types`, `feature_flags`, `segments`, `rollout_statuses`, `flag_sets`, `groups`, `users`, and `change_requests`.
- `split_io.py` implements the transport against the Admin API (`https://api.split.io/internal/api/v2`) with Bearer auth via `make_tracked_session()`, tenacity retries honoring `Retry-After` on 429, and the API's three pagination styles: limit/offset (`objects` envelope), marker-based (`data` envelope with `nextMarker`, used by `users` and `flag_sets`), and plain-array responses.
- Workspace-scoped endpoints fan out over all workspaces, injecting a `_workspace_id` column into every row. `feature_flags` and `segments` use a composite `["name", "_workspace_id"]` primary key since their names are only unique per workspace.
- Resumable syncs persist the next page URL plus current workspace, and save state after yielding each batch, so a heartbeat-timeout crash resumes mid-fan-out instead of restarting.
- Everything is full refresh: the Admin API exposes no server-side timestamp filter (timestamps are epoch-millisecond integers with no range params), so no endpoint declares `supports_incremental`.
- `validate_credentials` probes `/workspaces` at source-create (accepting 403 there, since a valid key may lack scope for unselected endpoints) and per-endpoint paths for schema checks. 401/403 are registered as non-retryable errors.
- Adds `canonical_descriptions.py` from the official API reference and sets `lists_tables_without_credentials = True` so the public docs render the table catalog.
- Moves `split_io` from Scaffolded to Implemented in `SOURCES.md`, and regenerates `SplitIoSourceConfig` (the `generated_configs.py` diff is scoped to that one class).

The source stays behind `unreleasedSource=True` with `releaseStatus=ALPHA` for now.

> [!NOTE]
> Endpoint pagination styles, paths, and param names are verified against the current Split Admin API reference (the OpenAPI spec embedded in docs.split.io) and cross-checked with Airbyte's connector manifest: `users`, `groups`, `flag_sets`, and `change_requests` are marker-paginated (`after` = `nextMarker`), `flag-sets` is served from `/internal/api/v3`, and the workspace-scoped lists use offset + `totalCount`. What I couldn't do is run authenticated requests (no Admin API key), so live behavior should be smoke-tested before promoting past alpha. Defenses if the docs are wrong anyway: envelope-key fallback between `objects` and `data`, offset advances by actual row count and terminates on `totalCount` (a server-side `limit` clamp can't skip rows), and an identical-page guard stops any endpoint whose server ignores pagination params instead of looping forever.

A user-facing doc for posthog.com (`contents/docs/cdp/sources/split-io.md`, matching the `docsUrl` slug) was written per the docs template and passes `audit_source_docs` for this source. It needs to land in the posthog.com repo separately.

## How did you test this code?

- `pytest products/warehouse_sources/backend/temporal/data_imports/sources/split_io/tests/` (71 passed): transport tests cover pagination termination for all three styles (including the server-clamped-limit and ignored-marker edge cases), workspace fan-out row injection and URL construction, resume-from-saved-state (skip completed workspace, mid-workspace URL), and credential probe/status mapping. Source tests cover schema listing, 403-at-create vs 403-per-schema validation, resume manager binding, and pipeline plumbing. No existing test exercised any of these paths since the source was a stub.
- `pytest products/warehouse_sources/backend/temporal/data_imports/sources/tests/` (1353 passed) for registry-wide checks (categories, generated configs).
- `ruff check` / `ruff format` clean on changed files. `makemigrations --check` shows no pending migration (the enum landed with the scaffold). `hogli ci:preflight` isn't available in this environment.
- Live API: verified base URL and auth error shape with curl (401 JSON envelope); could not verify authenticated pagination without an Admin key (see note above).

## Automatic notifications

- [ ] Publish to changelog?
- [ ] Alert Sales and Marketing teams?

## Docs update

posthog.com doc written (see note in Changes) but must land in the posthog.com repo.

## 🤖 Agent context

**Autonomy:** Human-driven (agent-assisted)

- Authored with PostHog Code (Claude). Skills invoked: `/implementing-warehouse-sources`, `/documenting-warehouse-sources`, `/writing-tests`.
- Modeled the implementation on the LaunchDarkly source (same shape: feature flag platform, parent fan-out, epoch-ms timestamps, full refresh only) rather than the generic `rest_source.RESTClient` path, because Split mixes three pagination styles across endpoints and needs workspace fan-out with resume state, which the plain-requests pattern handles more directly.
- Chose `PAGE_SIZE=50` over the documented max of 200 since per-endpoint caps vary and are unverifiable without credentials; termination is driven by `totalCount`, so a clamp only costs extra requests.
- Skipped the deep fan-out streams Airbyte also ships (`segments_keys`, per-environment flag definitions) - they need workspace x environment x segment iteration and can land as a follow-up once the base source is verified against a live account.

---
*Created with [PostHog Code](https://posthog.com/code?ref=pr)*

### CODE REVIEW

File: products/warehouse_sources/backend/temporal/data_imports/sources/split_io/settings.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        pagination="marker",
+        data_key="data",
+        # The users list is filtered by status; ACTIVE covers the account's usable members
+        # (matching other connectors for this API).
+        extra_params={"status": "ACTIVE"},
+    ),
+    "change_requests": SplitIoEndpointConfig(
+        name="change_requests",
+        path="/changeRequests",
+        primary_keys=["id"],
```
[the remark this answers is not in the document]
Comment (Gilbert09) [reply products/warehouse_sources/backend/temporal/data_imports/sources/split_io/settings.py]: Good catch — this was half right, and digging in surfaced two more issues. Verified against the current Split Admin API reference (the embedded OpenAPI spec on docs.split.io):

- `changeRequests` **is marker-paginated** (query params `status, limit, before, after, environmentId, objectType`; `nextMarker` in the response) — Airbyte's manifest uses offset there, but it's outdated. Switched to `pagination="marker"` in 09dfbbf.
- `groups` is also marker-paginated (`objects` envelope + `nextMarker`) — switched too.
- `flag-sets` is served from `/internal/api/v3`, not v2 — added a per-endpoint `api_version` and fixed the path.
- The current spec has no `status` param on the users list (responses include DEACTIVATED members), so the `status=ACTIVE` filter is dropped.

Also added the defensive guard you implied: if a server ignores our pagination params and returns an identical page twice, the paginator stops with a warning instead of looping/duplicating forever.

**Review by MarconLP — APPROVED** [review]

**Review by MarconLP — APPROVED** [review]
